Update USWDS, update gulp & stylesheet setup, update build process, update readme (#632)
usds/website: README.md

# USDS website

The USDS website is built with:

- [Jekyll](https://jekyllrb.com/) via the [github-pages gem](https://rubygems.org/gems/github-pages)
  - See [current dependencies](https://pages.github.com/versions/) for GitHub Pages
- [U.S. Web Design System (USWDS)](https://designsystem.digital.gov/)
- [Gulp](https://gulpjs.com/) / [USWDS Gulp](https://github.com/uswds/uswds-gulp)

Before getting started, install the following on your system:

- [Ruby](https://www.ruby-lang.org/en/documentation/installation/)
- [NPM](https://github.com/npm/cli)
- [Gulp CLI](https://gulpjs.com/docs/en/getting-started/quick-start)

## Install and run

### Initial setup
```
$ npm install
$ bundle install
$ npm start
```

### Running and building
#### Run locally & watch for changes
```
$ npm start
~ or ~
$ npm run serve
```

> Builds production Jekyll site, spins it up at `localhost:4000`, and watches for changes. Compiles main stylesheet with Gulp and watches for changes in `assets/stylesheets/**/*.scss`.

Reviewer: ```suggestion
Builds the Jekyll site, spins it up at `localhost:4000`, and watches for changes. Compiles main stylesheet with Gulp and watches for changes in `assets/stylesheets/**/*.scss`.
```

> #### Production build

Reviewer: ```suggestion
#### Local build
```

why: i'm worried this would be interpreted as `JEKYLL_ENV=production`, which has side effects of analytics tracking

Author: Oh nice callout. Pulling "production" out is a good idea here.

Had added that because _technically_ what gets built locally is what goes into production on GitHub pages. But we're also not using the production environment flag specifically.

```
$ npm run build
```

>  Builds production Jekyll site and compiles main stylesheet with Gulp.  

Reviewer: ```suggestion
 Builds local dev version of the Jekyll site and compiles main stylesheet with Gulp.  
```
